src/dune/link_time_code_gen.ml

```diff
@@ -33,20 +33,8 @@ let generate_and_compile_module cctx ~precompiled_cmi ~name:basename ~lib ~code
        |> Option.value_exn |> Path.as_in_build_dir_exn
      in
      Build.write_file_dyn ml code);
-  let opaque =
-    Ocaml_version.supports_opaque_for_mli (Super_context.context sctx).version
```

Comment:
@diml why do we compile with opaque (whenever it's available) here?

Reply:
IIRC, it's because the mli was built with `-opaque` and so the implem needs to be built with `-opaque` as well.

Reply:
Okay, thanks. I added a comment.